primes.rb

```diff
@@ -1,6 +1,21 @@
 class Primes
   def self.sum_to(limit = 100)
     # TODO - add your prime number solution here...
-    "I'm working on it!"
-  end
-end
\ No newline at end of file
+    primes = (2..limit).select do |n|
+    	(2..Math.sqrt(n)).none? do |i|
+    		(n%i).zero?
+    	end
+    end
+
+    	sum = primes.inject { |sum, i| sum + i }
+	end
+end
+
+Primes.sum_to(100)
+Primes.sum_to(1000)
+Primes.sum_to(10000)
+
+#puts Primes.sum_to(2000000)
+#commented out the 2 million soloution as it takes a while to run but it does work
```

Comment:
That's the trade off we need to make for larger numbers. Your approach is creating temporary arrays with all of the intermediate values in, which _looks_ great in Ruby code but suffers from being a bit inefficient. You'd either need to uppack this to use less OO loops, or take a look at what `lazy` does in the Ruby 2 Enumberable library.